### PR TITLE
Added bots

### PR DESCRIPTION
**Pull request by AmichaiYe**
Added 
  delete_access_ket
  revoke_access_key
  delete_image_from_ecs_repo

### CODE REVIEW

File: bots/delete_access_key.py [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+from botocore.exceptions import ClientError
+import bots_utils
+
+EVENT_NAME = 'CreateAccessKey'
+DEFAULT_CLOUDTRAIL_LOOKUP_TIME_DIFF=5
+
+
+def run_action(boto_session, rule, entity, params):
+    # create iam resource
+    iam_resource = boto3.resource('iam')
```
Comment (doronzil) [inline bots/delete_access_key.py:20]: please use boto3.client

File: bots/revoke_access_key.py [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+import boto3
+from botocore.exceptions import ClientError
+import bots_utils
+
+
+EVENT_NAME = 'CreateAccessKey'
+
+def run_action(boto_session, rule, entity, params):
+    # create iam resource
+    iam_resource = boto3.resource('iam')
```
Comment (doronzil) [inline bots/revoke_access_key.py:19]: please use boto3.client

File: bots/delete_access_key.py [hunk 3]
```diff
@@ -0,0 +1,46 @@
+"""
```
Comment (doronzil) [inline bots/delete_access_key.py:1]: you didn't push the bots.md and the relevant permissions

File: bots/delete_access_key.py [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+
+def run_action(boto_session, rule, entity, params):
+    # create iam resource
+    iam_resource = boto3.resource('iam')
+
+    # look for event in cloudtrail
+    event = bots_utils.cloudtrail_event_lookup(boto_session, entity, EVENT_NAME)
+
+    # in case that the cloudtrail_event_lookup failed to find events
+    if event is None:
```
Comment (doronzil) [inline bots/delete_access_key.py:26]: please change to more informative error

File: bots/delete_image_from_ecs_repo.py [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+
+def run_action(boto_session, rule, entity, params, report_time):
+    # creating ecr client
+    ecr_client = boto3.client('ecr')
+
+    # looking for the event
+    event = bots_utils.cloudtrail_event_lookup(boto_session, entity, EVENT_NAME)
+
+    if event is None:
+        return "Error, returned 0 events"
```
Comment (doronzil) [inline bots/delete_image_from_ecs_repo.py:24]: please change to more informative error

File: bots/revoke_access_key.py [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+
+    # look for the event in cloudtrail
+    event = bots_utils.cloudtrail_event_lookup(boto_session, entity, EVENT_NAME)
+
+    # in case that the cloudtrail_event_lookup failed to find events
+    if event is None:
+        return "Error, returned 0 events"
```
Comment (doronzil) [inline bots/revoke_access_key.py:26]: please change to more informative error

**Review by doronzil — CHANGES_REQUESTED** [review]

File: bots/delete_access_key.py (hunk 1 above)
Comment (masterA11) [reply bots/delete_access_key.py:20]: the method i'm using is only from an boto3.resource

File: bots/revoke_access_key.py [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+            AccessKeyId=access_key_id,
+            Status='Inactive'
+        )
+
+        # create the text output
+        return f'user: {username} access key with id : {access_key_id} was revoked'
+
+    except ClientError as e:
+        # in case of an unexpected error
+        return f"Unexpected error: {e} \n"
```
Comment (RavidAtias) [inline bots/revoke_access_key.py:46]: Change " to '

Comment (RavidAtias) [reply bots/revoke_access_key.py:46]: And no need to use \n

**Review by RavidAtias — CHANGES_REQUESTED** [review]
Nice! just change " to ' and remove \n from the errors

**Review by doronzil — APPROVED** [review]

**Review by doronzil — CHANGES_REQUESTED** [review]
Please change the bots names to iam_revoke_access_key/ iam_delete_access_key/ ecs_delete_repository_image in both bots.md and bots filenames

**Review by doronzil — APPROVED** [review]

**Review by RavidAtias — APPROVED** [review]

**Pull request merged**